src/main/scala/org/neo4j/spark/service/Neo4jQueryService.scala

```diff
@@ -1,39 +1,77 @@
 package org.neo4j.spark.service
 
 import org.apache.spark.sql.SaveMode
-import org.apache.spark.sql.sources.{And, EqualTo, Filter, IsNull, Not, Or}
-import org.neo4j.cypherdsl.core.{Condition, Conditions, Cypher, Functions, PropertyContainer, StatementBuilder}
+import org.apache.spark.sql.sources.{And, Filter, Or}
 import org.neo4j.cypherdsl.core.renderer.Renderer
-import org.neo4j.spark.{Neo4jOptions, QueryType}
+import org.neo4j.cypherdsl.core._
 import org.neo4j.spark.util.Neo4jImplicits._
 import org.neo4j.spark.util.Neo4jUtil
+import org.neo4j.spark.{Neo4jOptions, NodeSaveMode, QueryType}
 
-import collection.JavaConverters._
+import scala.collection.JavaConverters._
 
 class Neo4jQueryWriteStrategy(private val saveMode: SaveMode) extends Neo4jQueryStrategy {
   override def createStatementForQuery(options: Neo4jOptions): String =
-    s"""UNWIND ${"$"}events AS event
+    s"""UNWIND ${"$"}events AS $BATCH_VARIABLE
        |${options.query.value}
        |""".stripMargin
 
-  override def createStatementForRelationships(options: Neo4jOptions): String = throw new UnsupportedOperationException("TODO implement method")
+  private def keywordFromSaveMode(saveMode: Any): String = {
+    saveMode match {
+      case NodeSaveMode.Overwrite | SaveMode.Overwrite => "MERGE"
+      case NodeSaveMode.ErrorIfExists | SaveMode.ErrorIfExists => "CREATE"
+      case NodeSaveMode.Match => "MATCH"
+      case _ => throw new UnsupportedOperationException(s"Source SaveMode $saveMode not supported")
+    }
+  }
+
+  override def createStatementForRelationships(options: Neo4jOptions): String = {
+    val relationshipKeyword = keywordFromSaveMode(saveMode)
+    val sourceKeyword = keywordFromSaveMode(options.relationshipMetadata.sourceSaveMode)
+    val targetKeyword = keywordFromSaveMode(options.relationshipMetadata.targetSaveMode)
+
+    val relationship = options.relationshipMetadata.relationshipType.quote()
+
+    val sourceLabels = options.relationshipMetadata.source.labels
```

Comment:
This "turn a string into a bunch of labels" should be factored out into a mini function which can be reused, for generating safe cypher from an input string.

Additionally -- not sure if this happens -- but I don't see that you catch labels which contain the \`\ character or spaces.

Consider unit testing with source/target labels that are messed up, like:


```
":Label`A:ANOTHER LABEL: (╯°□°）╯︵ ┻━┻ "
```